src/prefect/client/client.py

```diff
@@ -202,114 +196,66 @@ def _request(
-            raise
+        headers = {"Authorization": "Bearer {}".format(self.token)}
+        session = requests.Session()
+        retries = Retry(
+            total=6,
+            backoff_factor=1,
+            status_forcelist=[500, 502, 503, 504],
+            method_whitelist=["DELETE", "GET", "POST"],
+        )
+        session.mount("https://", HTTPAdapter(max_retries=retries))
+        if method == "GET":
+            response = session.get(url, headers=headers, params=params)
+        elif method == "POST":
+            response = session.post(url, headers=headers, json=params)
+        elif method == "DELETE":
+            response = session.delete(url, headers=headers)
+        else:
+            raise ValueError("Invalid method: {}".format(method))
+
+        # Check if request returned a successful status
+        response.raise_for_status()
+
+        return response
 
     # -------------------------------------------------------------------------
     # Auth
     # -------------------------------------------------------------------------
 
-    def login(
-        self,
-        email: str,
-        password: str,
-        account_slug: str = None,
-        account_id: str = None,
-    ) -> None:
+    def login(self, api_token: str) -> None:
         """
-        Login to the server in order to gain access
+        Logs in to Prefect Cloud with an API token. The token is written to local storage
+        so it persists across Prefect sessions.
 
         Args:
-            - email (str): User's email on the platform
-            - password (str): User's password on the platform
-            - account_slug (str, optional): Slug that is unique to the user
-            - account_id (str, optional): Specific Account ID for this user to use
+            - api_token (str): a Prefect Cloud API token
 
         Raises:
             - AuthorizationError if unable to login to the server (request does not return `200`)
         """
-
-        # lazy import for performance
-        import requests
-
-        # TODO: This needs to call the main graphql server and be adjusted for auth0
-        url = os.path.join(self.graphql_server, "login_email")  # type: ignore
-        response = requests.post(
-            url,
-            auth=(email, password),
-            json=dict(account_id=account_id, account_slug=account_slug),
-        )
-
-        # Load the current auth token if able to login
-        if not response.ok:
-            raise AuthorizationError("Could not log in.")
-        self.token = response.json().get("token")
-        if self.token:
-            creds_path = os.path.expanduser("~/.prefect/.credentials")
-            if not os.path.exists(creds_path):
-                os.makedirs(creds_path)
-            with open(os.path.join(creds_path, "auth_token"), "w+") as f:
-                f.write(self.token)
+        if not os.path.exists(self.local_token_path):
+            os.makedirs(self.local_token_path)
```

Comment:
So the `local_token_path` appears to include the filename as well -> have you verified that this `makedirs` call doesn't accidentally make a _directory_ with the name of the intended token file?

Reply:
Excellent call, no

Reply:
Fixed and added test!

Reply:
Thanks!